contracts/staking/test/migration.ts

```diff
@@ -122,15 +136,74 @@ blockchainTests('Migration tests', env => {
 
             it('reverts if init() reverts', async () => {
                 await enableInitRevertsAsync();
-                const tx = proxyContract.attachStakingContract.awaitTransactionSuccessAsync(initTargetContract.address);
+                const tx = proxyContract.attachStakingContract.awaitTransactionSuccessAsync(
+                    initTargetContract.address,
+                    constants.NULL_ADDRESS,
+                    constants.NULL_ADDRESS,
+                    constants.NULL_ADDRESS,
+                    constants.NULL_ADDRESS,
+                );
                 return expect(tx).to.revertWith(REVERT_ERROR);
             });
+
+            it('calls init with initialized addresses if passed in args are null', async () => {
+                const wethProxyAddress = randomAddress();
+                const ethVaultAddress = randomAddress();
+                const rewardVaultAddress = randomAddress();
+                const zrxVaultAddress = randomAddress();
+                await proxyContract.setAddressParams.awaitTransactionSuccessAsync(
+                    wethProxyAddress,
+                    ethVaultAddress,
+                    rewardVaultAddress,
+                    zrxVaultAddress,
+                );
+                const receipt = await proxyContract.attachStakingContract.awaitTransactionSuccessAsync(
+                    initTargetContract.address,
+                    constants.NULL_ADDRESS,
+                    constants.NULL_ADDRESS,
+                    constants.NULL_ADDRESS,
+                    constants.NULL_ADDRESS,
+                );
+                const logsArgs = filterLogsToArguments<TestInitTargetInitAddressesEventArgs>(
+                    receipt.logs,
+                    'InitAddresses',
+                );
+                for (const args of logsArgs) {
+                    expect(args.wethProxyAddress).to.eq(wethProxyAddress);
```

Comment:
Why are we just asserting `wethProxyAddress` and not the others?

Reply:
Whoops, forgot to add those in 😅